src/Sulu/Bundle/SecurityBundle/SingleSignOn/Adapter/OpenId/OpenIdSingleSignOnAdapter.php

```diff
@@ -261,7 +262,8 @@ private function createOrUpdateAdminUser(string $email, array $attributes): void
         }
 
         if (!\in_array($role->getIdentifier(), $roleNames, true)) {
-            $defaultRoleKey = new UserRole();
+            /** @var UserRole $defaultRoleKey */
+            $defaultRoleKey = $this->roleRepository->createNew();
```

Comment:
this looks wrong as we try to create a `UserRole` via `RoleRepository` which returns a `Role` not a `UserRole` which is wrong. This change I need be reverted:

```suggestion
            $defaultRoleKey = new UserRole();
```

The `UserRole` is not overwriteable and so not require the repository to be created.